src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -1230,7 +1230,8 @@ MulticopterPositionControl::control_non_manual(float dt)
 			} else {
 				// copter has reached our takeoff speed. split the thrust setpoint up
 				// into an integral part and into a P part
-				_thrust_int(2) = _takeoff_thrust_sp - _params.vel_p(2) * fabsf(_vel(2));
+				// remembering to remove _params.thr_hover which is added later as a feed-forward in control_position.
+				_thrust_int(2) = _takeoff_thrust_sp - _params.vel_p(2) * fabsf(_vel(2)) - _params.thr_hover;
 				_thrust_int(2) = -math::constrain(_thrust_int(2), _params.thr_min, _params.thr_max);
```

Comment:
As the integral is supposed to correct an inaccurate hover throttle parameter (after the feed forward was added), is there any reason why it should be constrained in exactly this way? Depending on whether the hover throttle parameter is slightly too large or to small, should it not be possible for the integral to be either positive or negative? Also note the sign in front of the constrained value, making the final value negative.